### PR TITLE
Tools: install aarch64 ARM compiler if required

### PR DESCRIPTION
**Pull request by peterbarker**
Co-authored-by: Peter Barker <pbarker@barker.dropbear.id.au>

### CODE REVIEW

File: Tools/environment_install/install-prereqs-ubuntu.sh [hunk 1]
```diff
@@ -158,27 +158,50 @@ fi
...
+    ARM_ROOT="gcc-arm-none-eabi-10-2020-q4-major"
+    case $(uname -m) in
+        x86_64)
+            if [ ! -d $OPT/$ARM_ROOT ]; then
+                (
+                    cd $OPT
+                    heading "Installing toolchain for STM32 Boards"
+                    echo "Installing toolchain for STM32 Boards"
+                    echo "Downloading from ArduPilot server"
+                    sudo wget https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2
```
Comment (tridge) [inline Tools/environment_install/install-prereqs-ubuntu.sh:172]: this should use ardupilot server

File: Tools/environment_install/install-prereqs-ubuntu.sh [hunk 2]
```diff
@@ -158,27 +158,50 @@ fi
...
+            echo "Done!";;
+
+        aarch64)
+            if [ ! -d $OPT/$ARM_ROOT ]; then
+                (
+                    cd $OPT
+                    heading "Installing toolchain for STM32 Boards"
+                    echo "Installing toolchain for STM32 Boards"
+                    echo "Downloading from ArduPilot server"
+                    sudo wget https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux.tar.bz2
```
Comment (tridge) [inline Tools/environment_install/install-prereqs-ubuntu.sh:192]: should use ardupilot server

**Review by tridge — CHANGES_REQUESTED** [review]
i've added gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux.tar.bz2 into the ArduPilot STM32-tools directory on firmware server

**Review by kenking2536 — COMMENTED** [review]
The modified code is tested and completed normally, which is more intelligent than the previous code,
But please be there as soon as possible https://firmware.ardupilot.org/Tools/STM32-tools/ Add gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux tar. Bz2 this file, otherwise downloading from ardupilot server cannot be realized. Use https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/ When installing this link, you will be unable to connect 3-4 times after 10 installations. You need to try again!

**Review by kenking2536 — CHANGES_REQUESTED** [review]
After adding files to the server, this code can be changed to the following:
# GNU Tools for ARM Embedded Processors
# (see https://launchpad.net/gcc-arm-embedded/)
ARM_ROOT="gcc-arm-none-eabi-10-2020-q4-major"
    case $(uname -m) in
        x86_64)
            if [ ! -d $OPT/$ARM_ROOT ]; then
                (
                    cd $OPT
                    heading "Installing toolchain for STM32 Boards"
                    echo "Installing toolchain for STM32 Boards"
                    echo "Downloading from ArduPilot server"
                    sudo wget https://firmware.ardupilot.org/Tools/STM32-tools/gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2
                    echo "Installing..."
                    sudo chmod -R 777 gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2
                    sudo tar xjf gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2
                    echo "... Cleaning"
                    sudo rm gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2
                )
            fi
            echo "Registering STM32 Toolchain for ccache"
            sudo ln -s -f $CCACHE_PATH /usr/lib/ccache/arm-none-eabi-g++
            sudo ln -s -f $CCACHE_PATH /usr/lib/ccache/arm-none-eabi-gcc
            echo "Done!";;

        aarch64)
            if [ ! -d $OPT/$ARM_ROOT ]; then
                (
                    cd $OPT
                    heading "Installing toolchain for STM32 Boards"
                    echo "Installing toolchain for STM32 Boards"
                    echo "Downloading from ArduPilot server"
                    sudo wget https://firmware.ardupilot.org/Tools/STM32-tools/gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux.tar.bz2
                    echo "Installing..."
                    sudo chmod -R 777 gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux.tar.bz2
                    sudo tar xjf gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux.tar.bz2
                    echo "... Cleaning"
                    sudo rm gcc-arm-none-eabi-10-2020-q4-major-aarch64-linux.tar.bz2
                )
            fi
            echo "Registering STM32 Toolchain for ccache"
            sudo ln -s -f $CCACHE_PATH /usr/lib/ccache/arm-none-eabi-g++
            sudo ln -s -f $CCACHE_PATH /usr/lib/ccache/arm-none-eabi-gcc
            echo "Done!";;
    esac

File: Tools/environment_install/install-prereqs-ubuntu.sh (hunk 1 above)
Comment (peterbarker) [reply Tools/environment_install/install-prereqs-ubuntu.sh]: Both wgets in here seem to work.  Not sure if someone dropped the tarball in there after seeing this...

**Comment by peterbarker** [comment]
Merged, thanks @kenking2536

**Pull request merged**